### PR TITLE
[Elasticsearch] Add metric type to fields

### PR DESCRIPTION
**Pull request by constanca-m**
## What does this PR do?

Adds metric type to the fields of the following data streams:
- [x] **CCR**:
  - Please check:
    - `elasticsearch.ccr.bytes_read`: counter or gauge?
    - `max_seq_no` is a gauge - Correct?
- [x] Cluster stats:
  - Everything is a gauge. There are no cumulative values.
- [x] Enrich:
  - Since there was no accurate way to test the metrics without the executing policy name defined, this will still need validation in the future.
- [x] Index
- [x] Index summary
- [x] Index recovery
- [x] Ingest pipeline
- [x] ML Job
- [x] Node
- [x] Node stats

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

Relates to https://github.com/elastic/integrations/issues/6618

### CODE REVIEW

**Comment by klacabane** [comment]
> enrich: Since there was no accurate way to test the metrics without the executing policy name defined, this will still need validation in the future.

Can we still use the value provided there even though it's not the right one ? What are the implications making the current value a blocker ?

**Comment by constanca-m** [comment]
> Can we still use the value provided there even though it's not the right one ?

Yes, of course. I just could not filter the value by the policy name, that is why I wrote that is still pending validation.

> What are the implications making the current value a blocker ?

Even if we set the metric type right away, the data stream cannot be migrated to TSDB because the dimensions are not defined. And the dimensions can only be set once we know the value of the policy name. @klacabane

**Comment by klacabane** [comment]
> Please check:
    elasticsearch.ccr.bytes_read: counter or gauge?
    max_seq_no is a gauge - Correct?

`elasticsearch.ccr.bytes_read`: gauge
`max_seq_no`: gauge

> And the dimensions can only be set once we know the value of the policy name

That's because the current value can be changing for a given policy, and we need a fixed one ?

File: packages/elasticsearch/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.8.0"
+  changes:
+    - description: Add metric type to metric fields for TSDB support.
```
Comment (ritalwar) [inline packages/elasticsearch/changelog.yml:4]:
```suggestion
    - description: Add metric_type mapping for the fields of metrics datasets to support TSDB.
```

**Review by ritalwar — APPROVED** [review]
LGTM!

**Pull request merged**